shell/platform/android/io/flutter/view/ResourceExtractor.java

```diff
@@ -47,48 +48,52 @@ private void extractResources() {
             }
 
             final AssetManager manager = mContext.getResources().getAssets();
-            try {
-                byte[] buffer = null;
-                final String[] assets = manager.list("");
-                LinkedList<String> assetList = new LinkedList<>(Arrays.asList(assets));
-                while(!assetList.isEmpty()) {
-                    String asset = assetList.pop();
-                    if (!mResources.contains(asset))
-                        continue;
-
-		    if (manager.list(asset).length > 0) {
-			// The asset is a directory
-			for (String a: manager.list(asset)) {
-			    assetList.add(asset + File.separator + a);
-			    mResources.add(asset + File.separator + a);
-			}
-			continue;
-		    }
 
+            byte[] buffer = null;
+            for (String asset : mResources) {
+                try {
                     final File output = new File(dataDir, asset);
-                    if (output.exists())
+
+                    if (output.exists()) {
                         continue;
+                    }
                     if (output.getParentFile() != null) {
                         output.getParentFile().mkdirs();
                     }
-                    try (InputStream is = manager.open(asset)) {
-                        try (OutputStream os = new FileOutputStream(output)) {
-                            if (buffer == null) {
-                                buffer = new byte[BUFFER_SIZE];
-                             }
-
-                            int count = 0;
-                            while ((count = is.read(buffer, 0, BUFFER_SIZE)) != -1) {
-                                os.write(buffer, 0, count);
+
+                    OutputStream os = null;
```

Comment:
This is [way too hard](https://romain.vernoux.fr/a-try-with-resources-equivalent-for-java-6-and-android/) and we really should be using try-with-resources here. See e.g. https://github.com/flutter/plugins/blob/master/packages/camera/android/src/main/java/io/flutter/plugins/camera/CameraPlugin.java#L545 for inspiration.

Reply:
The original code looks like it did the right thing (try-with-resources). Was there a reason for changing it?